File: factorial/factorial.py
def factorial(n: int) -> int:
    if n == 1:
        return 1
    return n * factorial(n - 1)

if __name__ == "__main__":
    print(factorial(1)) # 1
    print(factorial(2)) # 2 
    print(factorial(3)) # 6
    print(factorial(4)) # 24 
    print(factorial(5)) # 120

File: factorial/factorial.js
function factorial(n) {
  if (n === 1) {
    return 1;
  }

  return n * factorial(n - 1);
}

console.log(factorial(1));  // 1
console.log(factorial(2));  // 2
console.log(factorial(3));  // 6
console.log(factorial(4));  // 24
console.log(factorial(5));  // 120
